test/libyul/yulOptimizerTests/blockOutliner/basic.yul

```diff
@@ -0,0 +1,96 @@
+{
+	{
+		let a let b let c let d
+		{
+			a := mul(c,b)
+			if lt(a,c) {
+				a := add(a,c)
+			}
+		}
+		{
+			if eq(1,2) {
+				d := mul(a,b)
+				if lt(d,a) {
+					d := add(d,a)
+				}
+			}
+			{
+				d := add(d,a)
+			}
+		}
+		{
+			c := mul(a,b)
+			if lt(c,a) {
+				c := add(c,a)
+			}
+		}
+	}
+	function f(a, b) -> r {
+	    {
+			r := mul(a,b)
+			if lt(r, a) {
+			    r := add(r,a)
+			}
+		}
+		function g(x, y) -> z {
+		    z := mul(x,y)
+			if lt(z,x) {
+			    z := add(z,x)
+			}
+		}
+		r := g(b,a)
+	}
+	function h(a) -> r {
+		{
+		    r := add(r,a)
+		}
+		{
+		    r := add(r,a)
+		}
+	}
+}
+// ====
+// step: blockOutliner
+// ----
+// {
+//     {
+//         let a
+//         let b
+//         let c
+//         let d
+//         { a, b, c := _2(a, b, c) }
+//         {
+//             if eq(1, 2) { d, b, a := _2(d, b, a) }
+//             { d, a := _1(d, a) }
+//         }
+//         { c, b, a := _2(c, b, a) }
+//     }
+//     function f(a_1, b_2) -> r
+//     {
+//         {
+//             r, b_2, a_1 := _2(r, b_2, a_1)
+//         }
+//         function g(x, y) -> z
+//         { z, y, x := _2(z, y, x) }
+//         r := g(b_2, a_1)
+//     }
+//     function h(a_3) -> r_4
+//     {
+//         { r_4, a_3 := _1(r_4, a_3) }
+//         { r_4, a_3 := _1(r_4, a_3) }
+//     }
+//     function _1(a, c) -> a_4, c_5
```

Comment:
Can you use a prefix for function names?

Reply:
I was thinking about it, but I was wondering what to use... the function containing the block, if any, maybe? Especially since this combined with function inlining can replace equivalent function removal, but we don't want to loose the function names in the process... maybe something generic like ``outlinedBlock`` or even ``outlinedBlock_<sourceLocation>`` otherwise...

Reply:
Also the ``VarNameCleaner`` should probably be changed to treat function names just the same as variable names at some point.